src/Microsoft.Diagnostics.Monitoring.WebApi/Controllers/DiagController.cs

```diff
@@ -238,7 +238,7 @@ public Task<ActionResult<IEnumerable<Models.ProcessIdentifier>>> GetProcesses()
                         token => _dumpService.DumpAsync(processInfo.EndpointInfo, type, token),
                         egressProvider,
                         dumpFileName,
-                        processInfo.EndpointInfo,
+                        processInfo,
```

Comment:
Is there a reason not to follow this pattern in the `Result` method? Currently, it takes `IEndpointInfo`, but just skimming through the callers it looks like they're all doing `processInfo.EndpointInfo`. That would give you your `ProcessName` for all of those cases, unless I'm overlooking something.

Reply:
Most of them will work just fine, but there are a few calls that use `DumpAsync` that refer to endpoints retrieved through the `ICollectionRuleAction`, which seems to be locked into using endpoints (`newEndpointInfoTask` through a lot of the tests, a `CollectionRuleContext` inside `ActionListExecutor.cs`. The call I particularly found for this was in `CollectDumpAction.cs`).

We could change these ones in particular but then I'll have to add a 4th `EgressOperation` constructor. Thoughts?

Reply:
I think we are ok adding a 4th ctor, if it enables logs/metrics/trace/etc.